tests/pds/dataservice_test.go

```diff
@@ -745,6 +745,85 @@ var _ = Describe("{UpgradeDataServiceImage}", func() {
 	})
 })
 
+var _ = Describe("{DrainAndDecommissionNode}", func() {
+	JustBeforeEach(func() {
+		StartTorpedoTest("DrainAndDecommissionNode", "Deploys a data service, drains one selected node, decommissions that node", pdsLabels, 0)
+	})
+
+	It("Drain and Decommission a node", func() {
+		Step("Deploy, Validate, Drain Node, Cordon Node, Validate Data Service, Run Workload on Data Service, Uncordon Node", func() {
+			for _, ds := range params.DataServiceToTest {
+				isDeploymentsDeleted = false
+				deployment, _, _, err = DeployandValidateDataServices(ds, params.InfraToTest.Namespace, tenantID, projectID)
+				log.FailOnError(err, fmt.Sprintf("Error while deploying data services %s", ds.Name))
+				log.InfoD("Running Check Node DataService Func %v ", ds.Name)
+				nodes, err := pdslib.GetNodesOfSS(*deployment.ClusterResourceName, namespace)
+				log.FailOnError(err, "Cannot fetch nodes of the running Data Service")
+				var nodeName string
+				nodeName = nodes[0].Name // Selecting the 1st node in the list to cordon
+				Step("Drain Pods from a node", func() {
+					podsList, err := pdslib.GetPodsOfSsByNode(*deployment.ClusterResourceName, nodeName, namespace)
+					log.FailOnError(err, fmt.Sprintf("Pod not found on this Node : %s", nodeName))
+					log.InfoD("Pods found on %v node. Trying to Drain pods from this node now.", nodeName)
+					err = k8sCore.DrainPodsFromNode(nodeName, podsList, timeOut, maxtimeInterval)
+					log.FailOnError(err, fmt.Sprintf("Draining pod from the node %s failed", nodeName))
+					log.InfoD("Pods successfully drained from the node %s", nodeName)
+				})
+				Step("Cordon Selected Node", func() {
```

Comment:
Drain will internally cordon the node, in that case we dont need this step.

Reply:
Good catch. Addressed this comment by uncordoning the node right towards the end. Problem with drain is that it cordons and uncordons in the same method - so although we are utilising it we have to decommission this node for some time (maybe to start workload pods). This is the reason we are cordoning it , starting the workload and uncordoning it towards the end of TC.